core/src/main/scala/chisel3/internal/firrtl/IR.scala

```diff
@@ -567,7 +567,7 @@ private[chisel3] object ir {
 
   case class DefClass(id: Class, name: String, ports: Seq[Port], block: Block) extends Component
 
-  case class Circuit(
+  private[chisel3] case class Circuit(
```

Comment:
🎊 🎊 🎊

Reply:
I actually don't think this makes any difference lol. `object ir` is already package private so people already can't construct these--but the leakage of `Circuit` out via the annotations remains.